src/main/integrations/slack/index.ts

```diff
@@ -0,0 +1,78 @@
+import playerStateStore, { PlayerState, VideoDetails, VideoState } from "../../player-state-store";
+import IIntegration from "../integration";
+import { StoreSchema } from "~shared/store/schema";
+import Conf from "conf";
+import { PostType } from "../../@types/slack";
+
+export default class SlackStatus implements IIntegration {
+  private store: Conf<StoreSchema>;
+  private enabled = false;
+  private stateCallback: (event: PlayerState) => void = null;
+  private currentVideoId: string = "";
+  private statusTimeoutId: NodeJS.Timeout;
+  private slackUserToken: string = "";
+
+  private async playerStateChanged(state: PlayerState) {
+    if (!this.enabled) return;
+    const { secondsTillClearStatus } = await this.store.get("slack");
+    if (state.trackState === VideoState.Playing && state.videoDetails) {
+      if (!state.adPlaying) {
+        if (this.statusTimeoutId) clearTimeout(this.statusTimeoutId);
+        if (this.currentVideoId === state.videoDetails.id) {
+          this.statusTimeoutId = setTimeout(() => {
+            this.currentVideoId = "";
+            this.updateSlackStatus(true, state.videoDetails);
+          }, secondsTillClearStatus * 1000);
+        } else {
+          this.currentVideoId = state.videoDetails.id;
+          this.updateSlackStatus(false, state.videoDetails);
+        }
+      }
+    }
+  }
+  private updateSlackStatus(clear: boolean, data: VideoDetails) {
+    const currentUnixTimestamp = Math.floor(Date.now() / 1000);
+    const clearPostData: PostType = {
+      status_text: "",
+      status_emoji: "",
+      status_expiration: 0
+    };
+    const listeningToPostData: PostType = {
+      status_text: `Listening To: ${data.title} | ${data.author}`,
+      status_emoji: ":notes:",
+      status_expiration: currentUnixTimestamp + 15 * 60
+    };
```

Comment:
Can this become 1 variable instead?
I would play with the formatting a bit to possibly make it a bit more human readable, but something like this is what I am thinking about.
```js
const postData: PostType = (
   clear
   ? {
      status_text: "",
      status_emoji: "",
      status_expiration: 0
    }
    : {
      status_text: `Listening To: ${data.title} | ${data.author}`,
      status_emoji: ":notes:",
      status_expiration: currentUnixTimestamp + 15 * 60
    }
);

Reply:
its now instead of this, each keys value is set based on clear or not